dowhy/causal_refuters/dummy_outcome_refuter.py

```diff
@@ -101,117 +118,122 @@ def refute_estimate(self):
-
-                if callable(action):
-                    estimator = action(X_train, new_outcome_train, **func_args)
-                    new_outcome_train = estimator(X_train)
-                    new_outcome_validation = estimator(X_validation)
-                elif action in DummyOutcomeRefuter.SUPPORTED_ESTIMATORS:
-                    estimator = self._estimate_dummy_outcome(func_args, action, new_outcome_train, X_train)
-                    new_outcome_train = estimator(X_train)
-                    new_outcome_validation = estimator(X_validation)
-                elif action == 'noise':
-                    new_outcome_train = self._noise(new_outcome_train, func_args)
-                    new_outcome_validation = self._noise(new_outcome_validation, func_args)
-                elif action == 'permute':
-                    new_outcome_train = self._permute(new_outcome_train, func_args)
-                    new_outcome_validation = self._permute(new_outcome_validation, func_args)
-                elif action =='zero':
-                    new_outcome_train = np.zeros(new_outcome_train.shape)
-                    new_outcome_validation = np.zeros(new_outcome_train.shape)
-            
-            new_data = validation_df.assign(dummy_outcome=new_outcome_validation)
+        simulation_results = []
+        refute_list = []
+
+        for _ in self._num_simulations:
+            no_estimator = self.check_for_estimator()
+            if no_estimator:
+                X_train = np.zeros(self._data[self._chosen_variables].shape)
+                outcome_train = np.zeros(self._data['y'].shape)
+                X_validation = self._data[self._chosen_variables].values
+                outcome_validation = self._data['y'].values
+
+                outcome_validation = self.process_data(X_train, outcome_train, X_validation, outcome_validation, self._transformations)
+
+            groups = self.preprocess_data_by_treatment()
+            estimates = []
+            for key_train, _ in groups:
+                X_train = groups.get_group(key_train)[self._chosen_variables].values
+                outcome_train = groups.get_group(key_train)['y'].values
+                validation_df = []
+                transformations = self._transformations 
+                for key_validation, _ in groups:
+                    if key_validation != key_train:
+                        validation_df.append(groups.get_group(key_validation))
+
+                validation_df = pd.concat(validation_df)
+                X_validation = validation_df[self._chosen_variables].values
+                outcome_validation = validation_df['y'].values
+
+                # If the number of data points is too few, run the default transformation: [("zero",""),("noise", {'std_dev':1} )]
+                if X_train.shape[0] <= self._min_data_point_threshold:
+                    transformations = DummyOutcomeRefuter.DEFAULT_TRANSFORMATION

```

Comment:
why plural transformations? Are there more than one transformation?

Reply:
Well, the list can contain one or more transformations. To make it explicit, I'll just call is transformation_list.